static/js/services/message-contacts.js

```diff
@@ -34,9 +34,26 @@ angular.module('inboxServices').factory('MessageContacts',
         return value;
       });
       // populate the summaries of the result values then return the result
-      return GetContactSummaries(result).then(function() {
-        return result;
+      return GetContactSummaries(result);
+    };
+
+    var recordReadStatus = function(summaries) {
```

Comment:
This is the same code as in reports. Can we pull out / standardise this somewhere?